pandas/core/groupby/generic.py

```diff
@@ -1744,9 +1846,11 @@ def groupby_series(obj, col=None):
         if isinstance(obj, Series):
             results = groupby_series(obj)
         else:
+            # TODO: this is duplicative of how GroupBy naturally works
+            # Try to consolidate with normal wrapping functions
             from pandas.core.reshape.concat import concat
 
-            results = [groupby_series(obj[col], col) for col in obj.columns]
+            results = [groupby_series(content, label) for label, content in obj.items()]
             results = concat(results, axis=1)
```

Comment:
IIRC `concat` was only needed here (as opposed to just DataFrame constructor) because `results` could include DataFrames.  So it may no longer be needed